DAOS-10674 build: Add Ubuntu to build

jolivier23 opened the pull request:
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
DAOS-10674 build: Add Ubuntu to build

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try another option

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Add chmod

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try some more changes

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try again

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
TEst some stuff

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try again

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try something else

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 commented:
@brianjmurrell I may need some help figuring out what to change on this.

brianjmurrell commented on ucx.spec:
```diff
@@ -105,6 +105,7 @@ Provides header files and examples for developing with UCX.
            --disable-assertions \
            --disable-params-check \
            --without-java \
+           --prefix=%{_prefix} \
```
The `%configure` macro will include this itself.

```suggestion
```

brianjmurrell commented on ucx.spec:
```diff
@@ -121,8 +122,8 @@ Provides header files and examples for developing with UCX.
 %install
-make DESTDIR=%{buildroot} install
-rm -f %{buildroot}%{_libdir}/*.la
+make DESTDIR=%{buildroot} install prefix=%{_prefix} libdir=%{_libdir} datadir=%{_datadir}
```
These shouldn't be necessary.  IIUC the RPM builds were already working correctly so probably nothing in this file needs updating.

```suggestion
make DESTDIR=%{buildroot} install
```

brianjmurrell commented on ucx.spec:
```diff
@@ -121,8 +122,8 @@ Provides header files and examples for developing with UCX.
-make DESTDIR=%{buildroot} install
-rm -f %{buildroot}%{_libdir}/*.la
+make DESTDIR=%{buildroot} install prefix=%{_prefix} libdir=%{_libdir} datadir=%{_datadir}
+rm -f %{buildroot}%{_libdir}/*la
```
It really is only `.la` files we want to remove though, right?

```suggestion
rm -f %{buildroot}%{_libdir}/*.la
```

brianjmurrell commented on ucx.spec:
```diff
@@ -313,6 +314,9 @@ library internals, protocol objects, transports status, and more.
 %endif
 
 %changelog
+* Wed Jun 01 2022 Jeff Olivier <jeffrey.v.olivier@intel.com> - 1.12.1-2
+- Moved source to make file, add prefix
+
```
I don't think anything in the RPM build needs updating.

```suggestion
```

brianjmurrell commented on debian/patches/series:
```diff
@@ -0,0 +1,2 @@
+# static-bfd.patch
+undo-upstream.patch
```
I wonder if this patch really does get applied.  Our debian builds actually take their patches from the specfile.  I'm not sure if we will apply patches from this dir if they exist.  I will check the build.

So, yeah, this is the problem.  If you add this patch to the RPM specfile, things should work better.

I suppose we really ought to handle this situation better.

brianjmurrell committed:
Reverts

This reverts commits 962506d9192dbd56c96eccd33516c03f49b09cd4-abf43ec34aca8eb93d8623a4b2d27192d63ac4ff.

# NOTE: The following default set of pragmas will result in no build
#       or test.
#
#       Please edit the list of pragmas to produce the minimal amount of build
#       and test to prove your PR works.
#
#       Or if you have already done the above, and your PR is ready for it's
#       final for-landing run, remove all of the following pragmas.
#
#       More detail can be found at
#       https://wiki.hpdd.intel.com/display/CI/Commit+Pragmas.
Skip-checkpatch: true
Skip-build: true
# Or you can skip specific PR build stages:
#Skip-build-ubuntu20-rpm: true
#Skip-build-el7-rpm: true
#Skip-build-leap15-rpm: true
#Skip-build-centos7-gcc: true
#Skip-build-centos7-gcc-debug: true
#Skip-build-centos7-gcc-release: true
#Skip-build-leap15-gcc: true
#Skip-build-leap15-icc: true
#Skip-build-ubuntu-clang: true

# And/or you could just enable quick-build below which reduces build tims considerably
Quick-build: true 

Skip-unit-tests: true
# Or you could choose which Unit Test stages to skip below
#Skip-nlt: true
#Skip-unit-test: true	
#Skip-unit-test-memcheck: true
Skip-test: true 

# Or you can choose which test stages to skip below
#Skip-coverity-test: true
# Skip all Functional test stages
#Skip-func-test: true
# Skip Functional test stages on VMs
#Skip-func-test-vm: true
# Skip all Functional test stages on hardware
#Skip-func-hw-test: true
# or just specific hardware test stages
#Skip-func-hw-test-small: true
#Skip-func-hw-test-medium: true
#Skip-func-hw-test-large: true
#Skip-scan-centos-rpms: true
#Skip-test-centos-rpms: true

# If you do run any of the above Functional test stages, please use the following to
# limit your tests run to the tests/features you are working on
# You can add the tags for your tests to the appropriate pragmas below:
#Test-tag: -hw
#Test-tag-hw-small: hw,msall
#Test-tag-hw-medium: hw,medium,ib2
#Test-tag-hw-large: hw,large

# if you want to test your PR with a PR of a component, you can speify that PR:
#PR-repos: project@PR-number[:build_num]

# if you want to disable using RPMs in the Functional tests:
#RPM-test: false

# Or if you want to test with daos RPMs that are already built (i.e. in another PR or in the repo):
#RPM-test-version: version-release

# This should be figured out automatically, but you can force it with
#Doc-only: true

# If you are ready to push a full CI build and test you can uncomment
# the below pragma:
#Full-CI-run: true

jolivier23 committed:
Move files

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Add patch to spec

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Fix the patch

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

brianjmurrell committed:
Move undo-upstream.patch into specfile

So that our build system applies it.


#Pragmas from previous commit message:
Skip-checkpatch: true
Skip-build: true
Quick-build: true 
Skip-unit-tests: true
Skip-test: true 
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

jolivier23 committed:
Add not-installed files again

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

brianjmurrell committed:
Remove duplicate patch

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

jolivier23 committed:
Add commit to run tests

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Fix url

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Remove the source from make file

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Try something else

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Add new compat file

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
use debhelper

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Add files to not-installed

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

jolivier23 committed:
Upload new patch

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

brianjmurrell commented on debian/control:
```diff
@@ -42,7 +42,6 @@ Description: Header files for UCX library
 Package: libucx0
 Section: libs
 Depends: ${shlibs:Depends}, ${misc:Depends}
-Breaks: libopenmpi3 ( << 4.1.0-3)
```
I'd rather see this commented out with a comment that **we** did that, not upstream, and why.  Just that in the future, we know.

jolivier23 committed:
Add a comment

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>
